### PR TITLE
Optimise memory usage when reading V4 files.

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What

* Create group objects as each row is read instead of reading all lines
into memory first. This saves a significant amount of memory
* Use an alternative CSV parser that is more efficient. This reduced the CSV parsing time by ~30% and reduces CPU utilisation.

### How to review

Review changes, test.

### Who can review

Anyone

### CODE REVIEW

File: src/main/java/dp/xlsx/V4File.java [hunk 1]
```diff
@@ -25,57 +27,69 @@
...
 
-        final Map<Group, Group> groups = new HashMap<>();
+            CsvParserSettings settings = new CsvParserSettings();
+            settings.setEmptyValue("");
+            settings.setNullValue("");
+            CsvParser parser = new CsvParser(settings);
+
+            if ((line = bufferedReader.readLine()) != null) {
+
+                final String[] header = parser.parseLine(line);
```
Comment (Matt-Guest) [inline src/main/java/dp/xlsx/V4File.java]: This can return a null string array

File: src/main/java/dp/xlsx/V4File.java [hunk 2]
```diff
@@ -25,57 +27,69 @@
...
+                headerOffset = Integer.parseInt(v4Code.split("V4_")[1]) + 1;
+                headerGroup = new Group(header, headerOffset);
+            }
+
+            while ((line = bufferedReader.readLine()) != null) {
 
-            final Group group = new Group(row, headerOffset);
-            final String timeValue = row[headerOffset + 1];
-            final String observation = row[0];
+                final String[] row = parser.parseLine(line);
```
Comment (Matt-Guest) [inline src/main/java/dp/xlsx/V4File.java]: This can return a null string array

File: src/main/java/dp/xlsx/V4File.java (hunk 1 above)
Comment (CarlHembrough) [reply src/main/java/dp/xlsx/V4File.java]: I have updated the header parsing code to throw an exception if the header is null

File: src/main/java/dp/xlsx/V4File.java (hunk 2 above)
Comment (CarlHembrough) [reply src/main/java/dp/xlsx/V4File.java]: I have skipped the line if its parse result is null

**Pull request merged**